doc/connecting-lab.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Connecting LAVA Lab to the pipeline instance"
+date: 2024-02-19
+description: "Connecting a LAVA lab to the KernelCI pipeline"
+weight: 2
+---
+
+As we are moving towards the new KernelCI API and pipeline, we need to make sure
+all the existing LAVA labs are connected to the new pipeline instance.  This
+document explains how to do this.
+
+## Token setup
+
+The first step is to generate a token for the lab.  This is done by the lab admin,
+and the token is used to submit jobs from pipeline to the lab, and to authenticate
+LAVA lab callbacks to the pipeline.
+Requirements for the token:
+- Description: a string matching the regular expression `[a-zA-Z0-9\-]+`, for example "kernelci-new-api-callback"
+- Value: arbitrary, kept secret
+
+## Pipeline configuration
+
+### Secrets (toml) file
+
+The next step is to add the token to the pipeline services configuration files.
+Secrets are stored in the `kernelci.secrets` section of the `kernelci.toml` file,
```

Comment:
I'm a bit confused: should the token be placed in `kernelci.toml` under section:

```toml
[kernelci.secrets]
token=$token
```

or like described below? Please let me know where I could catch up with the most recent config structure if I'm missing well-known info that should be obvious to the reader.

Reply:
This whole story with section naming leaving me puzzled as well. I think kernelci.secrets is more relevant to kci cli tool and rework related to kernelci-core settings module. lava-callback doesn't use it, as it will be overkill for it (it should be lightweight as much as possible).
scheduler-lava doesn't use it too, so in this particular configuration for lava-callback i am using part of existed configuration scheme for scheduler-lava, and just added option for separate callback token (optional), or we can be reusing single token for both, requests and callback (this is what we did with lab-broonie).

Reply:
ACK, let's leave `kernelci.secrets` out of the scope of this review.